tests/test_waittask.py

```diff
@@ -0,0 +1,101 @@
+from asyncio import ensure_future
+import asyncio
+from time import time
+
+import pytest
+from syncer import sync
+
+from pyppeteer.errors import NetworkError, ElementHandleError, BrowserError
+
+
+@sync
+async def test_page_waitfor_selector_or_xpath(isolated_page, server):
+    p = isolated_page
+    # wait for selector
+    await p.goto(server / 'grid.html')
+    assert await p.waitFor('div')
+    # wait for xpath
+    assert await p.waitFor('//div')
+    # should not work for for single-slash xpaths
+    await p.setContent("""<div>some text</div>""")
+    with pytest.raises(ElementHandleError):
+        assert await p.waitFor('/div')
+
+
+@sync
+async def test_waitfor_multiline_body(isolated_page, server):
+    p = isolated_page
+    result = await p.waitForFunction("(() => true)()")
+    assert await result.jsonValue()
+
+
+@sync
+async def test_waitfor_predicate(isolated_page, server):
+    p = isolated_page
+    await asyncio.gather(
+        p.waitFor('() => window.innerWidth < 100'),
+        p.setViewport({'width': 10, 'height': 10})
+    )
+    # with args
+    await p.waitFor("(arg1, arg2) => arg1 !== arg2", "{}", 1, 2)
+
+
+@sync
+async def test_waitfor_unknown(isolated_page, server):
+    p = isolated_page
+    with pytest.raises(BrowserError, match='Unsupported target type'):
+        await p.waitFor({'foo': 'bar'})
+
+
+@sync
+async def test_waitfor_function(isolated_page, server):
+    p = isolated_page
+    # should work with string functions
+    watchdog = p.waitForFunction('window.__FOO === 1')
+    await p.evaluate('() => window.__FOO = 1')
```

Comment:
No need to wrap this 
```suggestion
    await p.evaluate('window.__FOO = 1')
```